app/assets/javascripts/components/AccountMenu.tsx

```diff
@@ -0,0 +1,984 @@
+      return;
+    }
+    setStatus(STRING_GENERATING_REGISTER_KEYS);
+    setIsAuthenticating(true);
+
+    const response = await application.register(
+      email as string,
+      password as string,
+      isEphemeral,
+      shouldMergeLocal
+    );
+
+    const error = response.error;
+    if (error) {
+      setStatus(undefined);
+      setIsAuthenticating(false);
+
+      application.alertService.alert(error.message);
+    } else {
+      setIsAuthenticating(false);
+      closeAccountMenu();
+    }
+  };
+
+  const handleAuthFormSubmit = (event:
+                                  TargetedEvent<HTMLFormElement> |
+                                  TargetedMouseEvent<HTMLButtonElement> |
+                                  TargetedKeyboardEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+
+    if (!email || !password) {
+      return;
+    }
+
+    blurAuthFields();
+
+    if (showLogin) {
+      login();
+    } else {
+      register();
+    }
+  };
+
+  const handleHostInputChange = (event: TargetedEvent<HTMLInputElement>) => {
+    const { value } = event.target as HTMLInputElement;
+    setUrl(value);
+    application.setHost(value);
+  };
+
+  const handleKeyPressKeyDown = (event: KeyboardEvent) => {
+    if (event.key === 'Enter') {
+      handleAuthFormSubmit(event as TargetedKeyboardEvent<HTMLButtonElement>);
+    }
+  };
+
+  const handlePasswordChange = (event: TargetedEvent<HTMLInputElement>) => {
+    const { value } = event.target as HTMLInputElement;
+    setPassword(value);
+  };
+
+  const handleEmailChange = (event: TargetedEvent<HTMLInputElement>) => {
+    const { value } = event.target as HTMLInputElement;
+    setEmail(value);
+  };
+
+  const handlePasswordConfirmationChange = (event: TargetedEvent<HTMLInputElement>) => {
+    const { value } = event.target as HTMLInputElement;
+    setPasswordConfirmation(value);
+  };
+
+  const handleMergeLocalData = async (event: TargetedEvent<HTMLInputElement>) => {
+    const { checked } = event.target as HTMLInputElement;
+
+    if (!checked) {
+      setShouldMergeLocal(checked);
+
+      const confirmResult = await confirmDialog({
+        text: STRING_ACCOUNT_MENU_UNCHECK_MERGE,
+        confirmButtonStyle: 'danger'
+      });
+      setShouldMergeLocal(!confirmResult);
+    }
+  };
+
+  const openPasswordWizard = () => {
+    closeAccountMenu();
+    application.presentPasswordWizard(PasswordWizardType.ChangePassword);
+  };
+
+  const openSessionsModal = () => {
+    closeAccountMenu();
+    appState.openSessionsModal();
+  };
+
+  const getEncryptionStatusForNotes = () => {
+    const length = notesAndTagsCount;
+    return `${length}/${length} notes and tags encrypted`;
+  };
+
+  const enableProtections = () => {
+    application.clearProtectionSession();
+
+    // Get the latest the protection status
+    setProtectionsDisabledUntil(getProtectionsDisabledUntil());
+  };
+
+  const refreshEncryptionStatus = useCallback(() => {
```

Comment:
This one should also be called on app launch event

Reply:
Done